packages/ui/src/models/camel-entities/SourceSchemaConfig.ts

```diff
@@ -0,0 +1,52 @@
+import { Schema } from '../schema';
+import { SourceSchemaType } from './SourceSchemaType';
+
+interface ISourceSchema {
+  schema: Schema | undefined;
+  multipleRoute: boolean;
+}
+
+interface IEntitySchemaConfig {
+  [SourceSchemaType.Route]: ISourceSchema;
+  [SourceSchemaType.Kamelet]: ISourceSchema;
+  [SourceSchemaType.Pipe]: ISourceSchema;
+  [SourceSchemaType.KameletBinding]: ISourceSchema;
+  [SourceSchemaType.Integration]: ISourceSchema;
+}
+
+class SourceSchemaConfig {
+  config: IEntitySchemaConfig = {
+    [SourceSchemaType.Route]: {
+      schema: undefined,
+      multipleRoute: true,
+    },
+    [SourceSchemaType.Kamelet]: {
+      schema: undefined,
+      multipleRoute: false,
+    },
+    [SourceSchemaType.Pipe]: {
+      schema: undefined,
+      multipleRoute: false,
+    },
+    [SourceSchemaType.KameletBinding]: {
+      schema: undefined,
+      multipleRoute: false,
+    },
+    [SourceSchemaType.Integration]: {
+      schema: undefined,
+      multipleRoute: true,
+    },
+  };
+
+  setSchema(name: string, schema: Schema) {
+    const type: SourceSchemaType = SourceSchemaType[name];
+    if (name === 'camelYamlDsl') {
+      this.config[SourceSchemaType.Route].schema = schema;
+    }
+    if (type) {
+      this.config[type].schema = schema;
+    }
+  }
```

Comment:
I saw the `TS` issue that you mentioned and I made a few changes to the method:

```typescript
  setSchema(name: string | SourceSchemaType, schema: Schema) {
    if (name === 'camelYamlDsl') {
      this.config[SourceSchemaType.Route].schema = schema;
      return;
    }

    if (isEnumType(name, SourceSchemaType)) {
      const type: SourceSchemaType = SourceSchemaType[name];
      this.config[type].schema = schema;
    }
  }
```

The `isEnumType` function is a special type of function, a Typeguard function, now called [`Type predicate`](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

It's defined like the following:
```typescript
function isEnumType<T extends object>(type: unknown, enumObject: T): type is keyof T {
  return typeof type === 'string' && (type as keyof typeof enumObject) in enumObject;
}
```

You could use that function and place it in a utils folder, maybe alongside of `packages/ui/src/utils/is-enum-type.ts`

Reply:
Many thanks!